src/util/cert/libcrypto/cert.c

```diff
@@ -176,31 +177,116 @@ errno_t sss_cert_pem_to_der(TALLOC_CTX *mem_ctx, const char *pem,
 #define IDENTIFIER_NISTP384 "nistp384"
 #define IDENTIFIER_NISTP521 "nistp521"
 
+static int sss_ec_get_key(BN_CTX *bn_ctx, EVP_PKEY *cert_pub_key,
+                          EC_GROUP **_ec_group, EC_POINT **_ec_public_key)
+{
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+    int ret;
+    char curve_name[4096];
+    unsigned char pubkey[4096];
+    size_t len;
+    EC_GROUP *gr;
+    EC_POINT *pk;
+
+    ret = EVP_PKEY_get_utf8_string_param(cert_pub_key,
+                                         OSSL_PKEY_PARAM_GROUP_NAME,
+                                         curve_name, sizeof(curve_name), &len);
+    if (ret != 1) {
+        ret = EINVAL;
+        goto done;
+    }
+
+    gr = EC_GROUP_new_by_curve_name(OBJ_sn2nid(curve_name));
+    if (gr == NULL) {
+        ret = EINVAL;
+        goto done;
+    }
+
+    ret = EVP_PKEY_get_octet_string_param(cert_pub_key,
+                                          OSSL_PKEY_PARAM_PUB_KEY,
+                                          pubkey, sizeof(pubkey), &len);
+    if (ret != 1) {
+        EC_GROUP_free(gr);
+        ret = EINVAL;
+        goto done;
+    }
+
+    pk = EC_POINT_new(gr);
+    if (pk == NULL) {
+        EC_GROUP_free(gr);
+        ret = EINVAL;
+        goto done;
+    }
+
+    ret = EC_POINT_oct2point(gr, pk, pubkey, len, bn_ctx);
```

Comment:
I guess this works somehow but IIUC `bn_ctx` is NULL here (see one of previous comments)?